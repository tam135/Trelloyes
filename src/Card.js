import React from 'react';
import './Card.css';

function Card(props) {
  return (
    <div className='Card'>
        <button
          type = 'button'
          onClick = {() => props.onClickDelete(props.id)}
          >
          Delete</button>
        <h3>{props.title}</h3>
        <p>{props.content}</p>
    </div>
  );
}
Card.propTypes = {
  onClickDelete: () => { }
}

export default Card;