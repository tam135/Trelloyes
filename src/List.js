import React from 'react';
import Card from './Card'
import './List.css'

function List(props) {
  return (
    <section className='List'>
        <header class="List-header">
            {props.header}
        </header>
        <div class="List-cards">
            {props.cards.map((card) => 
                <Card
                key = {card.id}
                title = {card.title}
                content = {card.content}
                />
            )}
            <button type="button">
                  + Add Random card
            </button>
        </div>
    </section>
  );
}

export default List;