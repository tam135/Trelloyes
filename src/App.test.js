import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';
import List from './List';

describe('Card component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Card name="Card"  />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the UI as expected with no unreads', () => {
        const tree = renderer
            .create(<Card name="Card" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('List component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<List name="List"  />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the UI as expected with no unreads', () => {
        const tree = renderer
            .create(<List name="List"  />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});