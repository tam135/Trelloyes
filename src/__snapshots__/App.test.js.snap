// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    Delete
  </button>
  <h3 />
  <p />
</div>
`;

exports[`Card component renders the UI as expected with no unreads 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    Delete
  </button>
  <h3 />
  <p />
</div>
`;
